### PR TITLE
bpo-34938: mimetypes.init() does not overwrite globals

### PR DESCRIPTION
**Pull request by AWhetter**
When importing mimetypes.init() into a module scope with
any of encodings_map, suffix_map, types_map, or common_types,
subsequent calls to init() would update the maps in
the local module scope and not in the mimetypes module.
This change makes the original maps always get updated.

This will break the undocumented behaviour of any changes to `mimetypes.encodings_map`, `mimetypes.suffix_map`, `mimetypes.types_map`, or `mimetypes.common_types` persisting after calling `mimetypes.init()` because the maps and their equivalent `mimetypes._default<map>` variable are now two separate dictionaries.

https://bugs.python.org/issue34938

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).



## CLA Missing

Our records indicate the following people have not signed the CLA:

@AWhetter

For legal reasons we need all the people listed to sign the CLA before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Comment by brandtbucher** [comment]
Thanks for your time @AWhetter, and welcome to CPython! 😎

I assume you've seen the bot's message about the CLA?

**Review by brandtbucher — CHANGES_REQUESTED** [review]
This looks like a good fix. I think it also needs a regression test in `Lib/test/test_mimetypes.py`.

Looking at the failed test, though, it seems that the current behavior of creating new objects rather than updating the existing ones is expected (though I *think* I agree with you that it's flawed).

Maybe someone from @python/email-team can shed some light on the desired behavior here?

**Comment by AWhetter** [comment]
I have seen the CLA notice. I've already signed up so I'm waiting for the approval to go through  the system at the moment.

Thanks for the fast feedback! I'll get a test added!

**Comment by zooba** [comment]
I posted on the bpo issue, as my comments ended up at a higher level than simple PR feedback.

Also, no need to force push changes. We'll squash merge at the end, and force pushing only breaks the review system.

**Review by maxking — APPROVED** [review]
LGTM

**Comment by Mariatta** [comment]
@maxking is this PR ready to merge? 

@zooba can you check the question in https://bugs.python.org/issue34938#msg359070, do you expect those to be worked on this PR, or separate? 

Thanks.

**Comment by AWhetter** [comment]
I've made the suggested documentation changes. The bit about importing `inited` into a local module might require more information. I'm tempted to add a section about it under `/faq/programming.rst` because it could have use elsewhere. What do we think?

**Review by YoSTEALTH — CHANGES_REQUESTED** [review]
There is no need to have line: 388-392 and line: 571 in `Lib/mimetypes.py` is there? Like take out the `_default_mime_types` function and indent back the content. Thanks

File: Lib/mimetypes.py
Comment (zooba) [inline Lib/mimetypes.py]: These `global` statements and the function (and its call) can be removed.

**Review by YoSTEALTH — CHANGES_REQUESTED** [review]
these change must be reverted back, you need to have it as:
``suffix_map = _suffix_map_default.copy()`` not ``suffix_map = _suffix_map_default = {...}`` because they are suppose to be 2 separate variable. On https://github.com/python/cpython/blob/d8fb9e1df51746ab6e26bdb68ab85fd0c0dd042c/Lib/mimetypes.py#L366 ``suffix_map.clear()`` is called and if they are same variable it will be bad.

**Review by iritkatriel — CHANGES_REQUESTED** [review]
This PR has merge conflicts now.

**Pull request closed without merging**
